src/eos-updater-util.c

```diff
@@ -154,14 +211,147 @@ eos_updater_resolve_upgrade (EosUpdater  *updater,
   if (!ostree_parse_refspec (o_refspec, &o_remote, &o_ref, error))
     goto out;
 
-  ret = (o_remote && o_ref && *o_remote && *o_ref);
-
   shuffle_out_values (upgrade_remote, o_remote, NULL);
   shuffle_out_values (upgrade_ref, o_ref, NULL);
 
   if (booted_checksum)
     *booted_checksum = g_strdup (booted);
 
+  /* Get branch configuration data baseurl */
+  repo_config = ostree_repo_get_config (repo);
+  osgroup = g_strdup_printf ("remote \"%s\"", osname);
+  baseurl = g_key_file_get_string (repo_config, osgroup, "url", NULL);
+  if (!baseurl)
+    {
+      g_set_error (error, G_IO_ERROR, G_IO_ERROR_NOT_FOUND,
+                   "No url found for %s, cannot upgrade", osname);
+      goto out;
+    }
+
+  /* Get product identifier */
+  hw_descriptors = g_hash_table_new_full (g_str_hash, g_str_equal,
+                                          g_free, g_free);
+  if (g_file_test (DT_COMPATIBLE, G_FILE_TEST_EXISTS))
```

Comment:
I personally think the current way is a bit easier to read, as I would have to move `fp` declaration out of the `if` branches and use separate variables for each branch, adding more lines to the already incredibly long list of local vars this function has. Do you think your suggestion has a performance or memory usage advantage comparing with the current proposal?